connectors/sources/outlook.py

```diff
@@ -0,0 +1,916 @@
+    "has_attachments",
+    "attachments",
+]
+CALENDAR_FIELDS = [
+    "required_attendees",
+    "type",
+    "recurrence",
+    "last_modified_time",
+    "subject",
+    "start",
+    "end",
+    "location",
+    "organizer",
+    "body",
+    "has_attachments",
+    "attachments",
+]
+
+END_SIGNAL = "FINISHED"
+
+
+def ews_format_to_datetime(datetime, timezone):
+    """Change datetime format to user account timezone
+    Args:
+        datetime: Datetime in UTC format
+        timezone: User account timezone
+    Returns:
+        Datetime: Date format as user account timezone
+    """
+    if isinstance(datetime, exchangelib.ewsdatetime.EWSDateTime):
+        return (datetime.astimezone(pytz.timezone(str(timezone)))).strftime(
+            "%Y-%m-%dT%H:%M:%SZ"
+        )
+    elif isinstance(datetime, exchangelib.ewsdatetime.EWSDate) or isinstance(
+        datetime, date
+    ):
+        return datetime.strftime("%Y-%m-%d")
+    else:
+        return datetime
+
+
+class TokenFetchFailed(Exception):
+    """Exception class to notify that connector was unable to fetch authentication token from Graph API"""
+
+    pass
+
+
+class UsersFetchFailed(Exception):
+    """Exception class to notify that connector was unable to fetch users from Active Directory"""
+
+    pass
+
+
+class UnauthorizedException(Exception):
+    """Exception class unauthorized calls"""
+
+    pass
+
+
+class RootCAAdapter(requests.adapters.HTTPAdapter):
+    """Class to verify SSL Certificate for Exchange Servers"""
+
+    def __init__(self):
+        self.ssl_verify = False
+        self.ssl_ctx = None
+
+    def cert_verify(self, conn, url, ssl_certificate_file, cert):
+        super().cert_verify(
+            conn=conn, url=url, verify=self.ssl_verify, cert=self.ssl_ctx
+        )
+
+
+class ExchangeUsers:
+    """Fetch users from Exchange Active Directory"""
+
+    def __init__(self, ad_server, domain, exchange_server, user, password, ssl_enabled):
+        self.ad_server = ad_server
+        self.domain = domain
+        self.exchange_server = exchange_server
+        self.user = user
+        self.password = password
+        self.ssl_enabled = ssl_enabled
+
+    def _create_connection(self, server):
+        return Connection(
+            server=server,
+            user=self.user,
+            password=self.password,
+            client_strategy=SAFE_SYNC,
+            auto_bind=True,  # pyright: ignore
+        )
+
+    async def get_users(self):
+        warnings.filterwarnings("ignore")
+        ldap_domain_name_list = ["DC=" + domain for domain in self.domain.split(".")]
+        search_query = ",".join(map(str, ldap_domain_name_list))
+
+        status, _, response, _ = self._create_connection(
+            server=Server(host=self.ad_server)
+        ).search(
+            search_query,
+            SEARCH_FILTER_FOR_NORMAL_USERS,
+            attributes=["mail"],
+        )
+
+        status, _, response_for_admin, _ = self._create_connection(
```

Comment:
We use overwrite `status` variable here. The variable is declared twice and it's used for detecting errors. IMO we should have separate variables or add a check after normal users search. 

Also are there other scenarios where we fail? E.g. can status be 404 or sth else that indicates an error?

Reply:
Ah I see! need to rename `status`. 

> Also are there other scenarios where we fail? E.g. can status be 404 or sth else that indicates an error?

No, `status` basically returns something that can be a boolean, dict, etc. so there is a chance of 0 and 1, either we get a value  or not.

❗ Here, `status` is not referring to the response status code.

Reply:
Yes, that needs to be fixed.